Lib/test/test_asyncgen.py

```diff
@@ -386,6 +387,80 @@ def tearDown(self):
         self.loop = None
         asyncio.set_event_loop_policy(None)
 
+    def test_async_gen_operator_anext(self):
+        async def gen():
+            yield 1
+            yield 2
+        g = gen()
+        async def consume():
+            results = []
+            results.append(await operator.anext(g))
+            results.append(await operator.anext(g))
+            results.append(await operator.anext(g, 'buckle my shoe'))
+            return results
+        res = self.loop.run_until_complete(consume())
+        self.assertEqual(res, [1, 2, 'buckle my shoe'])
+        with self.assertRaises(StopAsyncIteration):
+            self.loop.run_until_complete(consume())
+
+    def test_async_gen_operator_aiter(self):
+        async def gen():
+            yield 1
+            yield 2
+        g = gen()
+        async def consume():
+            return [i async for i in operator.aiter(g)]
+        res = self.loop.run_until_complete(consume())
+        self.assertEqual(res, [1, 2])
+
+    def test_async_gen_operator_aiter_class(self):
+        loop = self.loop
+        class Gen:
+            async def __aiter__(self):
```

Comment:
I based this off the `class Gen` defined in the existing `test_async_gen_asyncio_03()` test. But based on your previous comment, should these both use `def` rather than `async def`?

Reply:
1. `__aiter__` can be a synchronous function `def` and must return an asynchronous iterator.

2. Whan `__aiter__` is declared as an asynchronous generator, it is still **a synchronous function**, that just returns an instance of an asynchronous generator when you call it (so (1) still stands)

Reply:
Thanks for explaining, but I'm still confused. After reading (1) it sounded like `__aiter__` should be rewritten, but after reading (2) it sounded like the code should stay written as-is (even though it seems more confusing this way).

Reply:
It's OK for it to stay written as is.

> After reading (1) it sounded like __aiter__ should be rewritten, but after reading (2) it sounded like the code should stay written as-is (even though it seems more confusing this way).

(2) is an edge case of (1); the case of `__iter__` returning an iterator and `__iter__` being a generator function is absolutely the same.